--- a/usr/lib/python3.10/site-packages/lithops/serverless/backends/k8s/config.py	2024-04-15 21:08:08.229207612 +0200
+++ b/usr/lib/python3.10/site-packages/lithops/serverless/backends/k8s/config.py	2024-04-29 18:56:31.095651848 +0200
@@ -76,10 +76,12 @@
     version: lithops_vX.X.X
     user: lithops-user
 spec:
-  activeDeadlineSeconds: 600
-  ttlSecondsAfterFinished: 60
+  # mig 14apr2024 - Patch by Miguel @ SCONTAIN. Doubling the timeouts
+  activeDeadlineSeconds: 1200
+  ttlSecondsAfterFinished: 120
   parallelism: 1
-  backoffLimit: 0
+  # mig 14apr2024 - Patch by Miguel @ SCONTAIN. Rerun if failed at most 6 times
+  backoffLimit: 6
   template:
     spec:
       restartPolicy: Never
@@ -103,13 +105,32 @@
               valueFrom:
                 fieldRef:
                   fieldPath: status.podIP
+              # mig 14apr2024 - Patch by Miguel @ SCONTAIN. SCONE related variables
+            - name: SCONE_HEAP
+              value: '768M'
+            - name: SCONE_MODE
+              value: 'AUTO'
+            - name: SCONE_ALLOW_DLOPEN
+              value: '2'
+            - name: SCONE_FORK
+              value: '1'
+            - name: SCONE_SYSLIBS
+              value: '1'
+            - name: SCONE_CAS_ADDR
+              value: '172.20.0.1'
+            - name: SCONE_LAS_ADDR
+              value: '172.20.0.1'
+            - name: SCONE_CONFIG_ID_TEST
+              value: 'Lithops-Benchmark-D41-123-45678-90120/benchmark'
           resources:
+            # mig 14apr2024 - Patch by Miguel @ SCONTAIN. Increased initial memory and cpu and memory limits
             requests:
               cpu: '0.2'
-              memory: 128Mi
+              memory: 4096Mi
             limits:
-              cpu: '0.2'
-              memory: 128Mi
+              cpu: '8'
+              memory: 8192Mi
+              sgx.k8s.io/sgx: "1"
       imagePullSecrets:
         - name: lithops-regcred
 """
